event-based-simulators/main/measurements.py

```diff
@@ -0,0 +1,136 @@
+            return
+        for data_dict in self.simulated_data:
+            log.info('Send create measurements requests')
+            base_dict = Measurements.create_extra_info_dict(self=self, data=data_dict)
+            measurement_dict = Measurements.create_individual_measurement_dict(self=self, data=data_dict)
+            base_dict.update(measurement_dict)
+            json_measurement_list.append(base_dict)
+        cumulocityAPI.create_measurements(self=self, measurements=json_measurement_list)
+        log.info("Finished create new measurements")
+
+    def create_extra_info_dict(self, data):
+        extraInfoDict = {
+            "type": f"{data.get('type')}",
+            "time": f"{data.get('time')}",
+            "source": {
+                "id": f"{self.device_id}"
+            }
+        }
+        return extraInfoDict
+
+    def create_individual_measurement_dict(self, data):
+        measurementDict = {
+            f"{data.get('type')}":
+                {
+                    f"{data.get('series')}":
+                        {
+                            "unit": f"{data.get('unit')}",
+                            "value": f"{data.get('value')}"
+                        }
+                }
+        }
+        return measurementDict
+
+    def generate_measurement(self):
+        for measurement_def in self.measurements_definition:
+            distribution = measurement_def.get("valueDistribution", "uniform")
+            value = 0.0
+            if (distribution == "uniform"):
+                min_value = measurement_def.get("minimumValue", measurement_def.get("value"))
+                max_value = measurement_def.get("maximumValue", measurement_def.get("value"))
+                value = round(uniform(min_value, max_value), 2)
+            elif (distribution == "uniformint"):
+                min_value = measurement_def.get("minimumValue", measurement_def.get("value"))
+                max_value = measurement_def.get("maximumValue", measurement_def.get("value"))
+                value = randint(min_value, max_value)
+            elif (distribution == "normal"):
+                mu = measurement_def.get("mu")
+                sigma = measurement_def.get("sigma")
+                value = round(gauss(mu, sigma), 2)
+            self.simulated_data.append({
+                'type': measurement_def.get("type"),
+                'series': measurement_def.get("series"),
+                'value': value,
+                'unit': measurement_def.get("unit"),
+                'time': f"{self.timestamp}"
+            })
+
+    def get_or_create_device_id(self):
+        sim_id = self.model['id']
+        label = self.model['label']
+        self.device_id = cumulocityAPI.get_or_create_device(sim_id, label)
+
+
+def load(filename):
+    try:
+        with open(filename) as f_obj:
+            return json.load(f_obj)
+    except Exception as e:
+        print(e, type(e))
+        return {}
```

Comment:
we have the same methods in event_based_simulators.py: is there a way to avoid such code duplication?